### PR TITLE
Payment service simplification refactor

### PR DESCRIPTION
**Pull request by mic-max**
## Changes

- Add `.dockerignore` to improve build times and image size
- Simplify away the basic error classes
- Merge `server.js` into `index.js`
- Simplify code, remove classes
- Update and set dependencies to fixed versions
- Working health check added

Showing healthy service
image

Note: for the above command to show healthy it needs the grpc_health_probe executable to be included in the image which is done by [this PR](https://github.com/open-telemetry/opentelemetry-demo-webstore/pull/93)

Output when starting payment service
image

Gracefully exit when using ctrl + c
image
With this code missing and we ctrl + c the docker compose the result of the running the command shows exited with non-zero value
image

Resources

- https://howchoo.com/devops/how-to-add-a-health-check-to-your-docker-container
- https://www.npmjs.com/package/grpc-health-check
- https://docs.docker.com/engine/reference/builder/#healthcheck

### CODE REVIEW

File: src/paymentservice/Dockerfile [hunk 1]
```diff
@@ -12,25 +12,26 @@
...
-COPY ./src/paymentservice/package.json ./
+COPY ./src/paymentservice/package*.json ./
 
-RUN npm install --only=production
+RUN npm ci --omit=dev
 
 # -----------------------------------------------------------------------------
 
-FROM base
+FROM node:16-alpine
```
Comment (mic-max) [inline src/paymentservice/Dockerfile:25]: note: `node:18-alpine` had build issues

**Review by julianocosta89 — CHANGES_REQUESTED** [review]
Hey @mic-max sorry for not taking a look at this one before.

I've tried it locally and it seems we have some issues.
The 1st thing is that I can't build the container without the `package-lock.json` file.
I had to generate it locally and run the `docker compose build` again.

I think in this case, to make the users' life easier, we could simply commit the `package-lock.json`. Just not sure how that would affect different OS.

---

Once the container was built, whenever running the sample I got the following error in the UI:
```shell
rpc error: code = Internal desc = failed to charge card: could not charge the card: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp: lookup paymentservice on 127.0.0.11:53: no such host"
failed to complete the order
main.(*frontendServer).placeOrderHandler
	/usr/src/app/handlers.go:393
main.instrumentHandler.func1
	/usr/src/app/middleware.go:120
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2047
go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux.traceware.ServeHTTP
	/go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux@v0.29.0/mux.go:145
github.com/gorilla/mux.(*Router).ServeHTTP
	/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210
main.(*logHandler).ServeHTTP
	/usr/src/app/middleware.go:94
main.ensureSessionID.func1
	/usr/src/app/middleware.go:148
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2047
net/http.serverHandler.ServeHTTP
	/usr/local/go/src/net/http/server.go:2879
net/http.(*conn).serve
	/usr/local/go/src/net/http/server.go:1930
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1581
```

Looks like the proto files were not loaded properly.

**Comment by julianocosta89** [comment]
I've tested after your last commit and now we are getting the following error on the paymentservice>
```shell
{"level":40,"time":1657173084895,"pid":1,"hostname":"4e4174c9376f","err":{"type":"TypeError","message":"Cannot read properties of undefined (reading 'setAttributes')","stack":"TypeError: Cannot read properties of undefined (reading 'setAttributes')\n    at Object.chargeServiceHandler [as charge] (/usr/src/app/index.js:31:10)\n    at handleUnary (/usr/src/app/node_modules/@grpc/grpc-js/build/src/server.js:688:13)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"},"msg":"Cannot read properties of undefined (reading 'setAttributes')"}
/usr/src/app/index.js:43
    span.recordException(err)
         ^

TypeError: Cannot read properties of undefined (reading 'recordException')
    at Object.chargeServiceHandler [as charge] (/usr/src/app/index.js:43:10)
    at handleUnary (/usr/src/app/node_modules/@grpc/grpc-js/build/src/server.js:688:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```


And the following error in the UI:
```shell
rpc error: code = Internal desc = failed to charge card: could not charge the card: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp: lookup paymentservice on 127.0.0.11:53: server misbehaving"
failed to complete the order
main.(*frontendServer).placeOrderHandler
	/usr/src/app/handlers.go:393
main.instrumentHandler.func1
	/usr/src/app/middleware.go:120
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2047
go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux.traceware.ServeHTTP
	/go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux@v0.29.0/mux.go:145
github.com/gorilla/mux.(*Router).ServeHTTP
	/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210
main.(*logHandler).ServeHTTP
	/usr/src/app/middleware.go:94
main.ensureSessionID.func1
	/usr/src/app/middleware.go:148
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2047
net/http.serverHandler.ServeHTTP
	/usr/local/go/src/net/http/server.go:2879
net/http.(*conn).serve
	/usr/local/go/src/net/http/server.go:1930
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1581
```

**Review by julianocosta89 — APPROVED** [review]
Awesome @mic-max!
Tested here and everything is looking good!
We should be good to go! 🚀

**Pull request merged**